tensorflow_datasets/structured/movielens.py

```diff
@@ -0,0 +1,422 @@
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""MovieLens dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import csv
+import os
+import textwrap
+import codecs
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@article{10.1145/2827872,
+author = {Harper, F. Maxwell and Konstan, Joseph A.},
+title = {The MovieLens Datasets: History and Context},
+year = {2015},
+issue_date = {January 2016},
+publisher = {Association for Computing Machinery},
+address = {New York, NY, USA},
+volume = {5},
+number = {4},
+issn = {2160-6455},
+url = {https://doi.org/10.1145/2827872},
+doi = {10.1145/2827872},
+journal = {ACM Trans. Interact. Intell. Syst.},
+month = dec,
+articleno = {19},
+numpages = {19},
+keywords = {Datasets, recommendations, ratings, MovieLens}
+}
+"""
+
+_DESCRIPTION = """
+This dataset describes 5-star rating from MovieLens,
+a movie recommendation service.
+Users were selected at random for inclusion.
+There are 5 versions included: 25m, latest-small, 100k, 1m, 20m.
+In all datasets, the movies data and ratings data are joined on "movieId".
+The 25m dataset, latest-small dataset, and 1m dataset have the same schema
+and they do not include demographic data.
+The 25m dataset is the latest stable version of the MovieLens dataset.
+It is recommended for research purposes.
+The latest-small dataset is a small dataset with
+the same schema as 25m and 20m.
+It is updated over time by GroupLens and its checksum might need updating.
+The 20m dataset is one of the most used MovieLens datasets in academic papers
+along with the 1m dataset.
+The 100k dataset and 1m dataset are the oldest MovieLens datasets.
+They both contain demographic data and have the same schema.
+"""
+
+_DATASET_OPTIONS = ['25m', 'latest-small', '100k', '1m', '20m']
+
+
+class MovieLens(tfds.core.GeneratorBasedBuilder):
+  """MovieLens rating dataset."""
+
+  # Using BuilderConfig directly since only the name field is needed for
+  # configuration.
+  BUILDER_CONFIGS = [
+      tfds.core.BuilderConfig(
+          name='25m',
+          description=textwrap.dedent("""\
+              This dataset contains 25000095 ratings across 62423 movies.
```

Comment:
Nit: For numeric values in comments, consider using consistent readable formatting (using three digit comma) in all values.
E.g., 25000095  -> 2,500,095

Reply:
Good point. Fixed in 333dc4fec316388b6440cb7baf64b755905c724e.